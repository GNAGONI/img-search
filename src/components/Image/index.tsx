import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import { CardActionArea } from '@mui/material';

type Props = {
  url: string;
  alt: string;
};

const Image = (props: Props) => (
  <Card sx={{ maxWidth: 345 }}>
    <CardActionArea>
      <CardMedia
        component="img"
        height="300"
        image={props.url}
        alt={props.alt}
      />
    </CardActionArea>
  </Card>
);

export default Image;
