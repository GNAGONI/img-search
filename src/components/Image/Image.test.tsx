import renderer from 'react-test-renderer';
import Image from './index';

it('renders correctly Image', () => {
  const tree = renderer
    .create(
      <Image url="www.123.com" alt="img" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});