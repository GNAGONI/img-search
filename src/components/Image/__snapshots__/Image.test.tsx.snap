// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Image 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="img"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="300"
      src="www.123.com"
    />
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
